akka-actor/src/main/scala/akka/actor/AbstractActor.scala

```diff
@@ -317,22 +317,22 @@ abstract class AbstractActor extends Actor {
 
 /**
  * If the validation of the `ReceiveBuilder` match logic turns out to be a bottleneck for some of your
- * actors you can consider to implement it at lower level by extending `UntypedAbstractActor` instead
+ * actors you can consider to implement it at lower level by extending `ClassicAbstractActor` instead
  * of `AbstractActor`. The partial functions created by the `ReceiveBuilder` consist of multiple lambda
  * expressions for every match statement, where each lambda is referencing the code to be run. This is something
  * that the JVM can have problems optimizing and the resulting code might not be as performant as the
- * untyped version. When extending `UntypedAbstractActor` each message is received as an untyped
+ * classic version. When extending `ClassicAbstractActor` each message is received as a classic
  * `Object` and you have to inspect and cast it to the actual message type in other ways (instanceof checks).
  */
-abstract class UntypedAbstractActor extends AbstractActor {
+abstract class ClassicAbstractActor extends AbstractActor {
```

Comment:
We can't rename this, because it would be binary and source incompatible.

Reply:
yes, this should remain as is

Reply:
Since it's only a helper, maybe deprecate old one and create a copy with the new name?

Reply:
@jrudolph I've done your suggestion so that we have progress vs just leaving relics, there are a few java tests that extend the `ClassicAbstractActor` and nothing using the `UntypedAbstractActor` but we have bin compat. WDYT?

Reply:
No we shouldn't change or add more things here. This is completely unrelated to what we lately have called untyped. This originates from the ancient https://github.com/akka/akka/blob/release-2.3/akka-actor/src/main/scala/akka/actor/UntypedActor.scala which was the original Java API for Actor.

Reply:
I see thanks. Reverting.

Reply:
> This originates from the ancient /akka-actor/src/main/scala/akka/actor/UntypedActor.scala@release-2.3 which was the original Java API for Actor.

Reply:
> This originates from the ancient /akka-actor/src/main/scala/akka/actor/UntypedActor.scala@release-2.3 which was the original Java API for Actor.

It's still somewhat confusing to just keep it. Is it still used or useful? Should we deprecate / remove it later? (Not in this PR)

Reply:
we can deprecate it, created issue https://github.com/akka/akka/issues/27630